main/acle.md

```diff
@@ -12375,7 +12451,27 @@ While (resulting in predicate tuple)
   // _b64[_s64]_x2, _b8[_u64]_x2, _b16[_u64]_x2, _b32[_u64]_x2 and
   // _b64[_u64]_x2
   svboolx2_t svwhilelt_b8[_s64]_x2(int64_t rn, int64_t rm);
-  ```
+```
+
+### SME2 lookup table
+
+#### MOVT
+
+Move vector register to ZT0.
+``` c
+  // Variants are also available for:
+  // [_s8], [_u16], [_s16], [_u32], [_s32], [_u64], [_s64]
+  //[_bf16], [_f16], [_f32], [_f64]
+  void svmovt_zt[_u8](uint64_t zt0, svuint8_t zt, uint64_t idx) __arm_streaming __arm_inout("zt0");
```

Comment:
`svwrite_lane_zt[_u8]` might fit better with the MOVA handling.  The problem with “mov” as a C/C++ function name is that it doesn't indicate the direction of the move, and so becomes ambiguous if moves in both directions are supported in future.

It might also be worth having:
```
  void svwrite_zt[_u8](uint64_t zt0, svuint8_t zt) __arm_out("zt0");
```
for the 0-index case, for which the incoming value of ZT0 is ignored.

Reply:
I agree that we should have  svwrite 
and the extra intrinsic for when index is zero.

But I am not sure we should remove the attribute __arm_streaming
I can see the pseudo code in:
https://developer.arm.com/documentation/ddi0602/2024-03/SME-Instructions/MOVT--vector-to-table---Move-vector-register-to-ZT0-?lang=en
has:
CheckStreamingSVEEnabled
CheckSMEZT0Enabled
I can see for the MOVAZ in sve2p1(https://developer.arm.com/documentation/ddi0602/2024-03/SME-Instructions/MOVAZ--tile-to-vector--two-registers---Move-and-zero-two-ZA-tile-slices-to-vector-registers-?lang=en ) it uses  __arm_streaming in the ACLE PR#309  https://github.com/ARM-software/acle/pull/309/files  and it has 
CheckStreamingSVEAndZAEnabled

Reply:
Agree, the difference in `zt0` handling ("inout" vs. "out") warrants another intrinsic (normally I'd object against purely convenience functions).